scrapy/pipelines/media.py

```diff
@@ -197,35 +197,32 @@ def _cache_result_and_execute_waiters(self, result, fp, info):
             defer_result(result).chainDeferred(wad)
 
     # Overridable Interface
+    @abstractmethod
     def media_to_download(self, request, info, *, item=None):
         """Check request before starting download"""
-        pass
+        raise NotImplementedError()
 
+    @abstractmethod
     def get_media_requests(self, item, info):
         """Returns the media requests to download"""
-        pass
+        raise NotImplementedError()
 
+    @abstractmethod
     def media_downloaded(self, response, request, info, *, item=None):
         """Handler for success downloads"""
-        return response
+        raise NotImplementedError()
 
+    @abstractmethod
     def media_failed(self, failure, request, info):
         """Handler for failed downloads"""
-        return failure
+        raise NotImplementedError()
 
+    @abstractmethod
     def item_completed(self, results, item, info):
         """Called per item when all media requests has been processed"""
-        if self.LOG_FAILED_RESULTS:
-            for ok, value in results:
-                if not ok:
-                    logger.error(
-                        "%(class)s found errors processing %(item)s",
-                        {"class": self.__class__.__name__, "item": item},
-                        exc_info=failure_to_exc_info(value),
-                        extra={"spider": info.spider},
-                    )
-        return item
```

Comment:
I think this (and the tests for it) should stay, in case some 3rd-party subclasses call super() for it (as bundled ones don't).